src/grpc_server.cc

```diff
@@ -3889,14 +3889,14 @@ ModelInferHandler::Process(InferHandler::State* state, bool rpc_ok)
           InferResponseComplete, reinterpret_cast<void*>(state));
     }
     // Get request ID for logging in case of error.
-    const char* request_id = nullptr;
+    const char* request_id = "";
     if (irequest != nullptr) {
       LOG_TRITONSERVER_ERROR(
           TRITONSERVER_InferenceRequestId(irequest, &request_id),
           "unable to retrieve request ID string");
     }
 
-    if ((request_id == nullptr) || (request_id[0] == '\0')) {
+    if ((!strcmp(request_id, "")) || (request_id[0] == '\0')) {
```

Comment:
For simplicity and readability, can we just do this instead?

```suggestion
    if (!std::strncmp(request_id, "", 1)) {
```

See godbolt example for equivalency, null terminator check is redundant: https://gcc.godbolt.org/z/3EaqbE64e